pallets/proposals/src/impls.rs

```diff
@@ -277,7 +278,62 @@ impl<T: Config> Pallet<T> {
             Ok::<(), DispatchError>(())
         })?;
 
-        Self::deposit_event(Event::NoConfidenceRoundVotedUpon(who, project_key));
+        Self::deposit_event(Event::NoConfidenceRoundVotedUpon(who.clone(), project_key));
+
+
+        //once the voting is complete check if the confidence vote could be auto finalized
+        //getting the total threshold required for the total confidence
+        let voting_no_confidence_threshold: BalanceOf<T> =
+            T::PercentRequiredForVoteNoConfidenceToPass::get().mul_floor(project.raised_funds);
+
+        //verifying whether the no confidence vote has passed the threshold if so then auto finalize it
+        if nay_vote >= voting_no_confidence_threshold {
+            let locked_milestone_percentage =
+                project.milestones.iter().fold(Percent::zero(), |acc, ms| {
+                    if !ms.1.is_approved {
+                        acc.saturating_add(ms.1.percentage_to_unlock)
+                    } else {
+                        acc
+                    }
+                });
+
+            let project_account_id = Self::project_account_id(project_key);
+
+            match project.funding_type {
+                FundingType::Brief | FundingType::Proposal => {
```

Comment:
We only want to enable automatic refunds for grants and proposals.

Briefs need to go through the dispute pallet

Reply:
done